clients/src/test/java/org/apache/kafka/test/MockSelector.java

```diff
@@ -109,8 +110,28 @@ public void send(Send send) {
 
     @Override
     public void poll(long timeout) throws IOException {
-        this.completedSends.addAll(this.initiatedSends);
+        completeInitiatedSends();
+        completeDelayedReceives();
+        time.sleep(timeout);
+    }
+
+    private void completeInitiatedSends() throws IOException {
+        for (Send send : initiatedSends) {
+            completeSend(send);
+        }
         this.initiatedSends.clear();
+    }
+
+    private void completeSend(Send send) throws IOException {
+        // Consume the send so that we will be able to send more requests to the destination
+        ByteBufferChannel discardChannel = new ByteBufferChannel(send.size());
```

Comment:
Yeah, we need it because of this check: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java#L104.